packages/svelte/src/compiler/phases/2-analyze/css/Selector.js

```diff
@@ -116,13 +126,14 @@ export default class Selector {
 			if (block.global) {
 				remove_global_pseudo_class(block.selectors[0]);
 			}
-			if (block.should_encapsulate)
+			if (block.should_encapsulate) {
 				encapsulate_block(
 					block,
-					index === this.blocks.length - 1
+					index === this.blocks.filter(block => !block.invisible).length - 1
```

Comment:
might be worth a comment or a temporary variable explaining what this is doing (though that was true before this PR!)